src/EpisodeManagementService/CreateUpdateEpisode/CreateUpdateEpisode.cs

```diff
@@ -34,21 +34,40 @@ public async Task<HttpResponseData> Run([HttpTrigger(AuthorizationLevel.Anonymou
                 _logger.LogInformation("PostData: {postData}", postData);
             }
         }
-        catch
+        catch (Exception ex)
         {
-            _logger.LogError("Could not read episode data.");
+            _logger.LogError("Could not read episode data. Error: {ex}", ex.Message);
             return req.CreateResponse(HttpStatusCode.BadRequest);
         }
 
         try
         {
-            var json = JsonSerializer.Serialize(episode);
-            await _httpRequestService.SendPost(Environment.GetEnvironmentVariable("CreateEpisodeUrl"), json);
-
-            return req.CreateResponse(HttpStatusCode.OK);
+            // Check if episode exists
+            var getEpisodeUrl = $"{Environment.GetEnvironmentVariable("GetEpisodeUrl")}?EpisodeId={episode.EpisodeId}";
+            var getEpisodeResponse = await _httpRequestService.SendGet(getEpisodeUrl);
+            if (getEpisodeResponse.StatusCode == HttpStatusCode.OK)
+            {
+                _logger.LogInformation("Episode {episodeId} already exists and will be updated.", episode.EpisodeId);
+                await _httpRequestService.SendPut(Environment.GetEnvironmentVariable("UpdateEpisodeUrl"), JsonSerializer.Serialize(episode));
+                _logger.LogInformation("UpdateEpisode function called successfully.");
+                return req.CreateResponse(HttpStatusCode.OK);
+            }
+            else if (getEpisodeResponse.StatusCode == HttpStatusCode.NotFound)
+            {
+                _logger.LogError("Episode {episodeId} does not exist and will be created.", episode.EpisodeId);
```

Comment:
This one should be LogInformation rather than LogError

Reply:
Changed log type to LogInformation.

Reply:
This is done now.